### PR TITLE
feat(sign-up): implement sign-up mutation

### PR DESCRIPTION
**Pull request by denistakeda**
issue #4

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ import express from 'express';
 import webpack from 'webpack';
 import { isDebug } from '../config/app';
 import { connect } from './db';
-import initPassport from './init/passport';
+//import initPassport from './init/passport';
```
Comment (sv3k) [inline server/index.js]: Maybe just remove this line?

File: server/schema/mutation/signUp.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  name: 'SignUpPayload',
+  fields: {
+    user: {type: userType},
+    token: {type: GraphQLString},
+  }
+});
+
+const signUp = {
+  args: {
+    name: {type: new GraphQLNonNull(GraphQLString)},
```
Comment (sv3k) [inline server/schema/mutation/signUp.js]: So we require not only an email/password but a user name as well. Isn't it too much for a simple todo app? 😃 I'd suggest to make it optional at least.

File: server/db/mongo/models/user.js [hunk 3]
```diff
@@ -15,17 +15,7 @@ import mongoose from 'mongoose';
...
-    name: { type: String, default: '' },
-    gender: { type: String, default: '' },
-    location: { type: String, default: '' },
-    website: { type: String, default: '' },
-    picture: { type: String, default: '' }
-  },
-  resetPasswordToken: String,
-  resetPasswordExpires: Date,
-  google: {}
+  name: { type: String },
```
Comment (sv3k) [inline server/db/mongo/models/user.js]: Maybe just `name: String`? Like for password field.

**Review by sv3k — COMMENTED** [review]
I suggest to think again if we really need the username as a mandatory field on sign up.

**Comment by denistakeda** [comment]
@sv3k Fixed, review again, please

An example mutation for test:

```
mutation signUp {
  signUp(
    email: "denis.takeda26@gmail.com",
    password: "Some password"
  ) {
    token
    user {
      id
      email
    }
  }
}
```

File: server/db/mongo/models/user.js [hunk 4]
```diff
@@ -15,7 +15,7 @@ import mongoose from 'mongoose';
 const UserSchema = new mongoose.Schema({
   email: { type: String, unique: true, lowercase: true },
   password: String,
-  name: { type: String },
+  name: String,
```
Comment (sv3k) [inline server/db/mongo/models/user.js]: Maybe we can remove this field from DB since it was removed from GraphQL layer.

Comment (denistakeda) [reply server/db/mongo/models/user.js]: I think that's ok. User can set this field later if he wants,

**Pull request merged**